File: Distributed_Systems/Threading/sleepy.py
#! python

# Sleepy class
"""Demonstrates timed behavior within a thread."""
import threading, time

class Sleepy( threading.Thread ):
	def __init__(self, data, count, delay):
		threading.Thread.__init__(self)
		self.data= data
		self.count= count
		self.delay= delay
		
	def run(self):
		self.delay_start()
		# loop -- print data count times
		for i in range(self.count):
			print( self.data )
			self.wait_for_it(2)

	def delay_start(self):
		self.wait_for_it(self.delay)

	def wait_for_it(self, wait_time):
		time.sleep(wait_time)

File: Distributed_Systems/Threading/Sleepy.java

public class Sleepy implements Runnable {
	String data;
	int count, delay_start;
	
	public Sleepy( String data, int count, int delay_start )
	{
		this.data= data;
		this.count= count;
		this.delay_start= delay_start;
	}

	public void run()
	{
		waitForIt( delay_start );
		for( int i= 0; i < count; i++ )
		{
			waitForIt( 2000 );
			printMessage();
		}
	}
	
	void waitForIt( int millis )
	{
		try
		{
			Thread.sleep( millis );
		}
		catch( Exception e )
		{
			System.err.println( "Couldn't sleep..." );
		}
	}
	
	void printMessage()
	{		
		System.out.println( data );
	}
}
